### DISCUSSION TITLE
Mobile support

### DISCUSSION THREAD

**Discussion by kbrede** [SebastianMC/obsidian-custom-sort#54] [General]
I did a search and couldn't find the answer. Is this plugin supposed to work with mobile? I've tried three different themes in case that prevented this working on mobile. I'm using an iPhone.

BTW, really enjoying this on my laptop!

Thanks,

Kent

**Comment by SebastianMC** [comment]
Hi @kbrede 

the most concise answer is: **yes, this plugin works also on mobile and is correctly ordering the folders and files**. This can be observed if your vault is shared (e.g. via some cloud storage) between desktop and mobile, by comparing the order of items in File Explorer.

That is the concise answer, yet there are some '_but_'s

I noticed that:
- on a large-screen mobile (iPad) where the UI is full-sized, thus the ribbon easily available in full size, no problems encountered. Ribbon icon of the custom sort plugin is visible, correctly shows the plugin state, allows enabling and disabling the sorting by tapping
- on a small-screen mobile (iPhone), the ribbon goes into some dynamic mobile mode, collapsed, invisible or partially visible. In result, the custom sort icon doesn't reflect the custom sorting state correctly. While it allows (by tapping) activation and deactivation of the custom sorting, visually it remains frozen.

To use the custom sort plugin on a small-screen mobile, enabling the _notifications of state changes_ (in plugin settings) could be helpful. You would have a visual feedback that the custom sorting has been applied.

Also, the Obsidian command palette being easily available (swipe down gesture on small-screen mobiles) allows for quick steering of the plugin via commands: **sort-on** and **sort-off**

I plan for two improvements for small-screen mobile devices:
1. Fix the issue with visual state of ribbon icon, so that it correctly reflects the custom sort state (as it does on iPad and on desktop)
2. Expose an additional mobile-specific setting: a checkbox which keeps the state change notifications enabled on mobile, even if disabled on desktop. This should be useful for cloud-shared vaults (like mine)

And a fundamental question: do the above clarifications help in your case or you are facing a different scenario?
**Please let me know**

thanks,
Seb

**Reply by kbrede** [reply]
Got it. I didn't realize I had to toggle the custom sort on in mobile. I just assumed the setting was synced from my laptop via iCloud. Thanks for the help and for the plugin!

Kent
